src/OpenDebugAD7/AD7DebugSession.cs

```
@@ -1667,6 +1693,69 @@ protected override void HandleModulesRequestAsync(IRequestResponder<ModulesArgum
+                        if (disassemblyStream.Read((uint)disassembleArguments.InstructionCount, enum_DISASSEMBLY_STREAM_FIELDS.DSF_ALL, out uint pdwInstructionsRead, prgDisassembly) == HRConstants.S_OK)
+                        {
+                            Debug.Assert(disassembleArguments.InstructionCount == pdwInstructionsRead);
+                            foreach (DisassemblyData data in prgDisassembly)
+                            {
+                                if (data.dwFlags.HasFlag(enum_DISASSEMBLY_FLAGS.DF_HASSOURCE))
+                                {
+                                    Debug.Fail("Warning: engine supports mixed instruction/source disassembly, but OpenDebugAD7 does not.");
+                                }
+                                DisassembledInstruction instruction = new DisassembledInstruction() {
+                                    Address = data.bstrAddress,
+                                    InstructionBytes = data.bstrCodeBytes,
+                                    Instruction = data.bstrOpcode,
+                                    Symbol = data.bstrSymbol
+                                };
+                                response.Instructions.Add(instruction);
+                            }
+                        }
+                        else
+                        {
+                            throw new AD7Exception(AD7Resources.Error_ReadDisassemblyStream);
+                        }
+                    }
+                    else
+                    {
+                        throw new AD7Exception(AD7Resources.Error_GetDisassemblyStream);
+                    }
+                }
+                else
+                {
+                    throw new AD7Exception(AD7Resources.Error_GetMemoryContext);
+                }
+                responder.SetResponse(response);
+            } catch (AD7Exception e) {
```

>AD7Exception [](start = 21, length = 12)

Probably want to catch all exceptions